lib/llvmopencl/VAL/CloneFunctionInto.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+#ifndef VAL_CLONEFUNCTIONINTO_H
+#define VAL_CLONEFUNCTIONINTO_H 1
+
+#include "_libclang_versions_checks.h"
+#include <llvm/Transforms/Utils/Cloning.h>
+
+inline template <typename VectorT>
+void CloneFunctionIntoAbs(llvm::Function *NewFunc,
```

Comment:
What is the reason for this wrapper? Please add a comment. Why a 'VAL' subdir? What does it stand for? Please remove the subdir and rename the header.

Reply:
>What is the reason for this wrapper?

VAL stands for "Version Abstraction Layer". As you see, the constructs are used from multiple files and places, so usage of the wrapper and placing it into an own header is justified. An own dir is created to keep VAL headers in a dedicated place to have a little bit less mess.

Reply:
OK. Please put this helper/wrapper function to LLVMUtils.h. If we get a lot of these files, let's then add a new directory. Thanks.

Reply:
Done.